src/atci.c

```diff
@@ -228,6 +228,36 @@ bool atci_param_get_uint(atci_param_t *param, uint32_t *value)
     return true;
 }
 
+bool atci_param_get_int(atci_param_t *param, int32_t *value)
+{
+    if (param->offset >= param->length)
+    {
+        return false;
+    }
+
+    char c;
+
+    *value = 0;
+
+    c = param->txt[param->offset];
+
+    if(c == '+')
+    {
+        param->offset++;
+        atci_param_get_uint(param, (uint32_t*)value);
+        return true;
+    }
+    else if(c == '-')
+    {
+        param->offset++;
+        atci_param_get_uint(param, (uint32_t*)value);
+        *value *= -1;
+        return true;
+    }
+
+    return atci_param_get_uint(param, (uint32_t*)value);
```

Comment:
Would you want to consolidate the above code into the following much shorter version?
```c
int mul = c == '-' ? -1 : 1;
if (c == '+' || c=='-') param->offset++;

bool rv = atci_param_get_uint(param, (uint32_t*)value);
*value *= mul;
return rv;

Reply:
Thanks, I was thinking a lot about how to optimize those conditions. Will integrate